### DISCUSSION TITLE
Blog article 《How TypeScript 4.9 `satisfies` Your Prisma Workflows》example error

### DISCUSSION THREAD

**Discussion by zwl1619** [prisma/prisma#22739] [Q&A] [answered]
There is an example in this article [《How TypeScript 4.9 `satisfies` Your Prisma Workflows》](https://www.prisma.io/blog/satisfies-operator-ur8ys8ccq7zb), as follow:

https://www.prisma.io/blog/satisfies-operator-ur8ys8ccq7zb#infer-the-output-type-of-methods-like-findmany-and-create
![图片5](https://github.com/prisma/prisma/assets/13177997/b36d45b0-8352-4b6f-a5f8-a7cbf36534f8)

I copied the typescript code and schema code into vscode, then ran `npx prisma generate`. Then vscode showed some errors, as follow:
![Snipaste_2024-01-22_14-55-09](https://github.com/prisma/prisma/assets/13177997/615c739d-c872-4bcf-a43b-71916b8ca638)

What should I do?

**Comment by nurul3101** [comment, marked as answer]
Hello @zwl1619 👋 

Thank you for creating this GitHub Discussion. I was able to reproduce this.

Can you try this code snippet in the `satisfiesTest` function and check if it fixes the issue for you?

```ts
  // Create a strongly typed `PostSelect` object with `satisfies`
  const postSelect = {
    title: true,
    createdAt: true,
    author: {
      select: {
        name: true,
        email: true,
      },
    },
  } satisfies Prisma.PostSelect;

  // Infer the resulting payload type
  type MyPostPayload = Prisma.PostGetPayload<{ select: typeof postSelect }>;

  // The result type is equivalent to `MyPostPayload | null`
  const post = await prisma.post.findUnique({
    where: { id: "3" },
    select: postSelect,
  });
```

**Reply by zwl1619** [reply]
It works, thank you!

**Reply by nurul3101** [reply]
Thanks for confirming! 🙏 
I'll add a PR to fix this in the blog.

**Comment by gap689** [comment]
The blog post is not changed at current time (Nov 22 2024). However I cannot find for the case where you select the posts from author. Also, what about nested relationships?

example:

```
const authorSelect = {
    name: true,
    email: true,
    posts: {
        select: { ... }
       }
} satisfies Prisma.AuthorSelect; 

```
is this correct? instead of:
```
    ...
    select: {
        posts: true,
        }
    ...
```
